mypy/subtypes.py

```diff
@@ -1210,6 +1210,12 @@ def visit_type_type(self, left: TypeType) -> bool:
                 return True
             if right.type.fullname() == 'builtins.object':
                 return True
+            item = left.item
+            if isinstance(item, TypeVarType):
+                item = item.upper_bound
+            if isinstance(item, Instance):
```

Comment:
There is a "TODO: handle metaclasses?" comment above that I think can be removed now